### DISCUSSION TITLE
tryCreateComponentFromExpression() return type

### DISCUSSION THREAD

**Discussion by dhrblav** [molstar/molstar#308] [Q&A] [answered]
Hi,
I am struggling with tryCreateComponentFromExpression()'s return type (in StructureBuilder class).
This return type seems to be: 
StateObjectSelector<SO.Molecule.Structure, StateTransformer<StateObject<any, StateObject<D = any, T extends StateObject.Type<string> = StateObject.Type<any>>.Type<any>>, StateObject<...>, any>>

I am trying to extract or build a StructureComponentRef out of the returned value. Is this possible, and if so, how?
Many thanks.

**Comment by dsehnal** [comment, marked as answer]
Hi, the components are managed by ``this.plugin.managers.structure.hierarchy``. You will need to find the component using the ref provided by the selector in ``this.plugin.managers.structure.hierarchy.current``.

**Reply by dhrblav** [reply]
Excellent! Thank you David.
